index.js

```diff
@@ -72,11 +72,15 @@ module.exports = {
 
   sassOptions: function () {
     var env  = process.env.EMBER_ENV;
-    var options = (this.app && this.app.options && this.app.options.sassOptions) || {};
-    var parentOption = (this.parent && this.parent.app && this.parent.app.options && this.parent.app.options.sassOptions) || {};
     var envConfig = this.project.config(env).sassOptions;
 
-    Object.assign(options, parentOption);
+    // *Either* use the options for an addon which is consuming this, *or* for
+    // an app which is consuming this, but never *both* at the same time. The
+    // special case here is `"dummy"`, which only applies when testing an addon.
+    var useApp = this.app.name === 'dummy' || this.app.name === this.project.name();
+    var options = useApp
+      ? (this.app && this.app.options && this.app.options.sassOptions) || {}
```

Comment:
Any reason not to do:

```js
var options = (
  (this.parent && this.parent.options && this.parent.options.sassOptions) ||
  (this.app && this.app.options && this.app.options.sassOptions) ||
  {}
);
```

That might allow you to remove the usage of `useApp`...

Reply:
That makes some sense and is worth testing – though, unfortunately, my test environment where I could reliably repro the bug behind this is now unavailable; I'll have to see if I can set up a smaller repro of the original problem.